### PR TITLE
Upgrade CAPV version and introduce h/w version in templates

### PR DESCRIPTION
**Pull request by srm09**
### What this PR does / why we need it
Thie patch:
- Upgrades the CAPV version to 1.5.0 which has support for Node IPAM, DHCP overrides and upgradable hardware version of VMs
- Exposes a parameter to provide a hardware version that will be passed to the `VSphereMachineTemplate` objects for cluster creation.

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### CODE REVIEW

**Comment by srm09** [comment]
@yastij FYI

File: providers/infrastructure-vsphere/v1.5.0/ytt/overlay.yaml [hunk 1]
```diff
@@ -299,6 +303,10 @@ spec:
       resourcePool: #@ data.values.VSPHERE_RESOURCE_POOL
       server: #@ data.values.VSPHERE_SERVER
       template: #@ data.values.VSPHERE_TEMPLATE
+#@ if data.values.VSPHERE_WORKER_HARDWARE_VERSION != "" and data.values.VSPHERE_WORKER_HARDWARE_VERSION != None:
+      #@overlay/match missing_ok=True
+      hardwareVersion: #@ data.values.VSPHERE_WORKER_HARDWARE_VERSION
```
Comment (wjun) [inline providers/infrastructure-vsphere/v1.5.0/ytt/overlay.yaml:308]: Is there any change required for cc based clusters?

File: packages/cluster-api-provider-vsphere/bundle/config/upstream/metadata.yaml [hunk 2]
```diff
@@ -36,3 +36,6 @@ releaseSeries:
   - major: 1
     minor: 4
     contract: v1beta1
+  - major: 1
+    minor: 5
+    contract: v1beta1
```
Comment (wjun) [inline packages/cluster-api-provider-vsphere/bundle/config/upstream/metadata.yaml:41]: Remove this line.

File: providers/infrastructure-vsphere/v1.5.0/ytt/overlay.yaml (hunk 1 above)
Comment (srm09) [reply providers/infrastructure-vsphere/v1.5.0/ytt/overlay.yaml:308]: I have observed some other missing items from the cc VSphereMachineTemplates, such as PCI devices, Ipv6 settings. I was hoping to raise an issue for this, as a catch up for all these missing items, and have a single PR bridge the gap between the two template types.

File: packages/cluster-api-provider-vsphere/bundle/config/upstream/metadata.yaml (hunk 2 above)
Comment (srm09) [reply packages/cluster-api-provider-vsphere/bundle/config/upstream/metadata.yaml]: This PR introduces a new minor version and hence the new entry.

**Review by wjun — APPROVED** [review]
LGTM.

**Pull request merged**